### DISCUSSION TITLE
[Question]About view's encapsulation

### DISCUSSION THREAD

**Discussion by suchuyang** [flet-dev/flet#3597] [Q&A]
### Question

I want to encapsulation some views class file because there's  many functions in each view. How can i do this job?

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment]
I dont seem to fully understand the question, but i think you could create a custom class: https://flet.dev/docs/getting-started/custom-controls

**Reply by suchuyang** [reply]
I write a example.Can you help me take a look and see if there are any issues.Thanks very much。

main.py

![main](https://github.com/flet-dev/flet/assets/18479786/1672e9db-eca4-4680-89a4-3000d50371df)

loginView.py
![loginn1](https://github.com/flet-dev/flet/assets/18479786/6f4c42ff-a7bf-47b8-a93a-cf1022132bda)
![login2](https://github.com/flet-dev/flet/assets/18479786/e3bb98c5-5e3e-4a71-8ae1-780bd49dfe15)
[loginExample.zip](https://github.com/user-attachments/files/16117897/loginExample.zip)

**Reply by ndonkoHenri** [reply]
The question is, do you face any issues? If yes, which...

**Reply by suchuyang** [reply]
Well ,I will keep trying. Thank you.

**Reply by vindevoy** [reply]
Check the following:

https://github.com/vindevoy/flet_mvc

I too did not like the fact that all my function that are triggered by user actions had to be stored within the main function.  Your application is more than one file ...

So I came up with the idea to use MVC and I wrote code to implement this. 

Note: this code serves only as an example, it's not production ready.
